scripts/module/money_drop.nut

```diff
@@ -0,0 +1,30 @@
+alive <- array(100, null);
+MoneyPickups <- {};
+
+function onPlayerSpawn( player ) {
+  alive[player.ID] = true;
+  player.Cash = 500;
+}
+
+function onPlayerPart( player, reason ) {
+  alive[player.ID] = null;
+}
+
+function onPlayerHealthChange( player, lastHP, newHP ) {
+  if ( newHP < 0.1 && alive[player.ID]) {
+    alive[player.ID] = false;
+    local pickup = CreatePickup( 337, player.World, player.Cash * 0.8, player.Position, 255, true);
+    if ( pickup ) {
+      MoneyPickups.rawset( pickup.ID, pickup);
+    }
+  }
+}
+
+function onPickupClaimPicked( player, pickup ) {
+  if ( alive[player.ID] ) {
+    MoneyPickups.rawdelete( pickup.ID );
+    pickup.Remove();
```

Comment:
If other scripts own pickups, you might accidentally delete them here.

You should make sure that this pickup belongs to `MoneyPickups` in the first place.